src/main/java/com/atex/ChangeAspectBeanType.java

```diff
@@ -0,0 +1,450 @@
+  // Input values
+  private static String cbAddress;
+  private static String cbBucket;
+  private static String cbBucketPwd;
+  private static String design;
+  private static String view;
+  private static boolean devView = false;
+  private static boolean dryRun = false;
+  private static int batchSize = -1;
+
+  private static Logger log = Logger.getLogger("Cleanup");
+
+  private static volatile Map<String, Long> totals = new TreeMap<>();
+
+  private static Bucket bucket;
+  private static String startKey;
+  private static int numThreads = 8;
+
+  private static volatile int processed = 0;
+  private static volatile int converted = 0;
+  private static volatile int removed = 0;
+
+  private static final Set<String> deletedKeys = Collections.synchronizedSet(new HashSet<>());
+  private static final Set<String> convertedKeys  = Collections.synchronizedSet(new HashSet<>());
+  private static int limit = -1;
+  private static int skip = -1;
+  private static volatile AtomicInteger lastPercentage = new AtomicInteger();
+  private static volatile AtomicLong lastTime = new AtomicLong();
+  private static int total = 0;
+  private static long timeStarted = 0;
+
+  private static boolean restore = false;
+  private static AtomicInteger restored = new AtomicInteger();
+
+  private static JsonDocument getItem(String id) {
+    JsonDocument response = null;
+    try {
+      response = bucket.get(id);
+    } catch (NoSuchElementException e) {
+      log.warning("No element with message: "
+          + e.getMessage());
+      e.printStackTrace();
+    }
+    return response;
+  }
+
+  private static boolean alreadyConverted(String id) {
+    synchronized (convertedKeys) {
+      if (convertedKeys.contains(id)) return true;
+      convertedKeys.add(id);
+      return false;
+    }
+  }
+
+  private static void execute() throws Exception {
+
+    String filename = "change-aspect-bean-type-" + new Date().getTime() + ".log";
+    FileHandler fileHandler = new FileHandler(filename);
+    SimpleFormatter simple = new SimpleFormatter();
+    fileHandler.setFormatter(simple);
+    log.addHandler(fileHandler);
+    log.setUseParentHandlers(false);
+
+    log.info ("Started @ " + new Date());
+    log.info("Couchbase node: "+cbAddress);
+
+    CouchbaseEnvironment env = DefaultCouchbaseEnvironment.builder()
+            .connectTimeout(TimeUnit.SECONDS.toMillis(60L))
+            .kvTimeout(TimeUnit.SECONDS.toMillis(60L))
+            .viewTimeout(TimeUnit.SECONDS.toMillis(1200L))
+            .maxRequestLifetime(TimeUnit.SECONDS.toMillis(1200L))
+            .autoreleaseAfter(5000)
+            .build();
+
+    Cluster cluster = CouchbaseCluster.create(env, cbAddress);
+
+    try {
+      bucket = cluster.openBucket(cbBucket, cbBucketPwd);
+    } catch (Exception e) {
+      cluster.authenticate("cmuser", cbBucketPwd);
+      bucket = cluster.openBucket(cbBucket);
+    }
+
+    process();
+
+    bucket.close();
```

Comment:
this should be in a finally block otherwise we may leave resources opened on the server.